### PR TITLE
Migrate from travis to GitHub actions

### LINKED ISSUE #28: Travis is broken
awesome-lint runs fine locally, but breaks on TravisCI with:

    Awesome list must reside in a valid git repository  remark-lint:awesome/github

https://travis-ci.org/RyanZim/awesome-npm-scripts/builds/536050660

### PR DESCRIPTION
**Pull request by mischah**
- Update dev dependencies
- Replace travis with GitHub action

fixes #28

### CODE REVIEW

**Comment by mischah** [comment]
This will fix #28 😗

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  lint:
+    name: Run awesome-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '18'
```
Comment (RyanZim) [inline .github/workflows/main.yml]: Might as well keep this evergreen:
```suggestion
          node-version: 'latest'
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: CI
+on:
+  - push
+  - pull_request
```
Comment (RyanZim) [inline .github/workflows/main.yml]: No need to run CI on pushes to non-master branches
```suggestion
  push:
    branches: master
  pull_request:
```

File: .github/workflows/main.yml (hunk 1 above)
Comment (mischah) [reply .github/workflows/main.yml]: Yup. That’s a very good idea.

File: .github/workflows/main.yml (hunk 2 above)
Comment (mischah) [reply .github/workflows/main.yml]: You’re right 👍🏻

**Comment by mischah** [comment]
Thanks for the review :octocat:

**Pull request merged**